### ISSUE TITLE
Add locally connected layer in keras

### ISSUE THREAD

**Issue by nitish11** [fchollet/keras#2589]
The DeepFace network architecture of Facebook [1] used locally connected layer to accelerate the training process while achieving very high classification accuracy.  Is there any implemented locally connected layer in Keras?

[1] Yaniv Taigman (Facebook), Ming Yang (Facebook), Marc'Aurelio Ranzato (Facebook), Lior Wolf (Tel Aviv University). DeepFace: Closing the Gap to Human-Level Performance in Face Verification. CVPR 2014, Columbus, Ohio.

**Comment by joelthchao** [comment]
I implemented it recently but haven't tested though. Not sure if Keras needs it, if so I may able to contribute some of my codes.

**Comment by nitish11** [comment]
@joelthchao : Could you please share the code link? 
Thanks in advance.

**Comment by joelthchao** [comment]
@nitish11 You can find it in
https://github.com/joelthchao/keras/blob/master/keras/layers/local.py
Notice that it's a draft version and only support 'mode=valid', please tell me if there is any error.

**Comment by nitish11** [comment]
@joelthchao : thanks.
Sure, I will do the experiments and let you know my observation.

**Comment by dvigneshwer** [comment]
Has any one tried to implement locally connected layer in keras?

**Comment by joelthchao** [comment]
@dvigneshwer it's in PR #2658.

**Comment by dvigneshwer** [comment]
Thanks @joelthchao

**Comment by alyato** [comment]
@nitish11,i meet this error when i use the joelthchao's code
![xxx](https://cloud.githubusercontent.com/assets/6815071/16517484/d9edf77a-3f3a-11e6-8cc5-f4bdd9b7930d.png)
DO you also meet this error,too.How do you solve it. Thanks.

**Comment by dvigneshwer** [comment]
Bleeding-edge version of Theano solves the above problem, try the following instructions below,

```
git clone git://github.com/Theano/Theano.git
cd Theano
python setup.py install

```

**Comment by nitish11** [comment]
The issue has been resolved in the commit   [835a02c](https://github.com/fchollet/keras/commit/835a02c037384b4ccf0e5b2425c6dd7469bf702d). 
Thanks @joelthchao :+1:
